velox/functions/prestosql/tests/RoundTest.cpp

```diff
@@ -59,30 +59,20 @@ class RoundTest : public functions::test::FunctionBaseTest {
   }
 
   template <typename T>
-  std::vector<std::tuple<T, int32_t, T>> testRoundWithDecFloatData() {
-    return {
-        {1.122112, 0, 1},
-        {1.129, 1, 1.1},
-        {1.129, 2, 1.13},
-        {1.0 / 3, 0, 0.0},
-        {1.0 / 3, 1, 0.3},
-        {1.0 / 3, 2, 0.33},
-        {1.0 / 3, 10, 0.3333333333},
-        {-1.122112, 0, -1},
-        {-1.129, 1, -1.1},
-        {-1.129, 2, -1.13},
-        {-1.129, 2, -1.13},
-        {-1.0 / 3, 0, 0.0},
-        {-1.0 / 3, 1, -0.3},
-        {-1.0 / 3, 2, -0.33},
-        {-1.0 / 3, 10, -0.3333333333},
-        {1.0, -1, 0.0},
-        {0.0, -2, 0.0},
-        {-1.0, -3, 0.0},
-        {11111.0, -1, 11110.0},
-        {11111.0, -2, 11100.0},
-        {11111.0, -3, 11000.0},
-        {11111.0, -4, 10000.0}};
+  std::vector<std::tuple<T, int32_t, T>> testRoundWithDecFloatAndDoubleData() {
+    return {{1.122112, 0, 1},         {1.129, 1, 1.1},
+            {1.129, 2, 1.13},         {1.0 / 3, 0, 0.0},
+            {1.0 / 3, 1, 0.3},        {1.0 / 3, 2, 0.33},
+            {1.0 / 3, 6, 0.333333},   {-1.122112, 0, -1},
+            {-1.129, 1, -1.1},        {-1.129, 2, -1.13},
+            {-1.129, 2, -1.13},       {-1.0 / 3, 0, 0.0},
+            {-1.0 / 3, 1, -0.3},      {-1.0 / 3, 2, -0.33},
+            {-1.0 / 3, 6, -0.333333}, {1.0, -1, 0.0},
+            {0.0, -2, 0.0},           {-1.0, -3, 0.0},
+            {11111.0, -1, 11110.0},   {11111.0, -2, 11100.0},
+            {11111.0, -3, 11000.0},   {11111.0, -4, 10000.0},
+            {0.575, 2, 0.58},         {0.574, 2, 0.57},
+            {-0.575, 2, -0.58},       {-0.574, 2, -0.57}};
```

Comment:
Can you add a test case for `round(57.499999999999993) == 57.0`?  That is also the behavior from Presto

Reply:
Yes, I verfied it on presto. I will check it on presto implementation.
![image](https://github.com/facebookincubator/velox/assets/14065538/6243e35f-4a5a-447e-ba17-1c9cf0581bac)

Reply:
Can you add this test in unit test in a separate PR?

Reply:
Actually can you try `round(0.49999999999999994, 0)`?